tests/PHPStan/Analyser/data/callsite-cast-narrowing.php

```diff
@@ -6,17 +6,21 @@
 
 class HelloWorld
 {
-	public function sayHello($mixed, int $int, string $string): void
+	/**
+	 * @param numeric-string $numericString
+	 * @param non-empty-string $nonEmptyString
+	 */
+	public function sayHello($mixed, int $int, string $string, $numericString, $nonEmptyString, bool $bool): void
 	{
 		if (ctype_digit((string) $mixed)) {
-			assertType('int<48, 57>|int<256, max>|numeric-string', $mixed);
+			assertType('int<0, max>|numeric-string|true', $mixed);
 		} else {
 			assertType('mixed', $mixed);
 		}
 		assertType('mixed', $mixed);
 
 		if (ctype_digit((int) $mixed)) {
-			assertType('int<48, 57>|int<256, max>|numeric-string', $mixed);
+			assertType('mixed', $mixed); // could be *NEVER*
```

Comment:
as can be seen in https://3v4l.org/k3Qce casting the parameter for `ctype-digit` to int beforehand does not yield a usefull result at all. 

therefore I adjusted the extension to only narrow on `(string)` call.

I think its not worth to add logic into the extension which turns all "bad uses" of casted parameters into a NeverType.

Reply:
https://github.com/phpstan/phpstan/issues/8736 is about `(string)` cast, not `(int)` cast, so please explain what changes for `(string)` that makes 1.9.13 not work.

Reply:
in 1.9.13 we assumed a parameters type before casting would be the same as the one the `ctype_digit` function narrows.

but because of this ugly edge case 
![grafik](https://user-images.githubusercontent.com/120441/213407011-39aa586b-1e62-4169-b55c-66296f34f0d9.png)
this is not entirely correct. you can actually pass in e.g. a `1` which casted to string is a numeric-string, but is not contained within the 

https://github.com/phpstan/phpstan-src/blob/a6a3441c829b81bb3630651dbb4664f41ff82649/src/Type/Php/CtypeDigitFunctionTypeSpecifyingExtension.php#L47-L57

type.

what we actually need is a type which represents all values which after casted with `(string)` will lead in a numeric-string.